src/examples/calculator.js

```diff
@@ -0,0 +1,255 @@
+  uses exact Decimal calculations instead of floating point. It shows how to
+  manipulate the DOM from the playground.`,
+  text: dedent`
+  // Make sure to click "Toggle DOM" above
+  // to see the app!
+
+  function isInteger(dec) {
+    const truncOptions = {
+      maximumFractionDigits: 0,
+      roundingMode: "down",
+    };
+    return Decimal.round(dec, truncOptions) === dec;
+  }
+
+  class Key extends React.Component {
+    render() {
+      const keyStyle = {
+        background: "gray",
+        borderRadius: "50%",
+        color: "white",
+      };
+      return (
+        <button style={keyStyle} onClick={() => this.props.onClick()}>
+          {this.props.children}
+        </button>
+      );
+    }
+  }
+
+  class Display extends React.Component {
+    render() {
+      const displayStyle = {
+        backgroundColor: "lightgray",
+        borderRadius: 5,
+        gridColumnEnd: "span 4",
+        fontSize: "2em",
+        fontFamily: "sans",
+        textAlign: "end",
+        padding: 5,
+      };
+
+      const { value, inputExponent } = this.props;
+      let display = value?.toString();
+
+      // The display should show a trailing decimal point
+      // and trailing zeroes if any have been entered.
+
+      // FIXME: This should instead be done with NumberFormat
+      // and its minimumFractionDigits option, but that is not
+      // yet implemented.
+      // FIXME: something weird about if-conditions: results
+      // are different when putting this expression directly
+      // in the if-statement
+      const cond = inputExponent !== 0m;
+      if (cond) {
+        if (isInteger(value)) {
+          display += ".";
+        }
+        const [, normalizedDecimalDigits] = display.split(".");
+        const missingZeroes =
+          -(Number(inputExponent) + 1) - normalizedDecimalDigits.length;
+        if (missingZeroes > 0) {
+          display += "0".repeat(missingZeroes);
+        }
+      }
+      return <div style={displayStyle}>{display}</div>;
+    }
+  }
+
+  class Calc extends React.Component {
+    static #initialState = {
+      // Decimal value where we accumulate typed digits
+      accum: 0m,
+      // If this is non-zero, the user entered a decimal
+      // point, and the value of the next typed digit
+      // will be multiplied by 10**inputExponent
+      inputExponent: 0m,
+      // Pending binary operation
+      op: null,
+      // First operand for the pending binary operation
+      store: null,
+      // If true, typing another digit will not add that
+      // digit to the accumulator, but instead clear the
+      // accumulator first. (This happens after typing
+      // the = key, or an operator)
+      nextDigitClears: false,
+    };
+
+    constructor() {
+      super();
+      this.state = Calc.#initialState;
+    }
+
+    #clear() {
+      this.setState(Calc.#initialState);
+    }
+
+    #appendDigit(digit) {
```

Comment:
Haha, I guess I originally didn't go that way because I wanted to use as much Decimal as possible, and then developed a blind spot around it... you're completely right that it'd be much more natural to implement the trailing-zero behaviour that way.